Makefile

```diff
@@ -1,56 +1,66 @@
-TOOLSDIR       := tools
-MWCCVER        := 2.0/sp1
-PROC           := arm946
+MWCCVER        := 2.0/sp2p2
+PROC           := arm946e
 PROC_S         := arm5te
 PROC_LD        := v5te
+LCF_TEMPLATE   := ARM9-TS.lcf.template
+LIBS           := -Llib -lsyscall -nostdlib
+ALL_BUILDDIRS  := $(BUILD_DIR)/lib
 
 include config.mk
 include common.mk
 include graphics_files_rules.mk
 include filesystem.mk
 
-ROM            := $(BUILD_DIR)/poke$(buildname).nds
-BANNER         := $(ROM:%.nds=%.bnr)
-BANNER_SPEC    := $(ROM:%.nds=%.bsf)
-ICON_PNG       := $(ROM:$(BUILD_DIR)/%.nds=icon/%.png)
+ROM             := $(BUILD_DIR)/poke$(buildname).nds
+BANNER          := $(ROM:%.nds=%.bnr)
+BANNER_SPEC     := $(buildname)/banner.bsf
+ICON_PNG        := $(buildname)/icon.png
+HEADER_TEMPLATE := $(buildname)/rom_header_template.sbin
 
 MWCFLAGS  += -ipa file $(DEFINES)
 MWASFLAGS += $(DEFINES)
 
-.SECONDARY:
-.SECONDEXPANSION:
-.DELETE_ON_ERROR:
-.PHONY: all tidy clean tools clean-tools sub
+.PHONY: main sub libsyscall
+.PRECIOUS: $(ROM)
 
 MAKEFLAGS += --no-print-directory
 
-all: $(ROM) $(NEF) $(ELF) $(SBIN)
+all: $(ROM)
 
 tidy:
-	$(MAKE) -C sub tidy
+	@$(MAKE) -C sub tidy
 	$(RM) -r $(BUILD_DIR)
 	$(RM) $(ROM)
 
-clean: tidy
+clean: tidy clean-tools
+	@$(MAKE) -C lib/syscall clean
+	@$(MAKE) -C sub clean
 
 main: $(SBIN)
```

Comment:
this depends on `tools`, will also depend on filesystem if we use NAIXs in the future